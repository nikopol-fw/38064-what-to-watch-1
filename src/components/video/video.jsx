// video.jsx

import React from 'react';
import PropTypes from 'prop-types';


export class Video extends React.PureComponent {
  constructor(props) {
    super(props);
    this._videoRef = React.createRef();
  }

  render() {
    const {src, preview, isPlaying} = this.props;

    return <video
      src={preview}
      poster={`/img/${src}`}
      muted={true}
      loop={true}
      preload={`metadata`}
      width={280}
      height={175}
      autoPlay={isPlaying}

      ref={this._videoRef}
    ></video>;
  }

  componentDidUpdate() {
    const video = this._videoRef.current;

    if (this.props.isPlaying) {
      video.play();
    } else {
      video.load();
    }
  }
}


Video.propTypes = {
  src: PropTypes.string.isRequired,
  preview: PropTypes.string.isRequired,
  isPlaying: PropTypes.bool.isRequired,
};
