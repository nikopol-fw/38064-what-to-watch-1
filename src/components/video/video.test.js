// video.test.js

import React from 'react';
import renderer from 'react-test-renderer';

import {Video} from './video';
import Mock from './../../mocks/films';


const mock = Mock[0];


it(`Video correctly renders`, () => {
  const isActive = true;

  const tree = renderer
    .create(<Video
      src={mock.preview}
      preview={mock.src}
      isPlaying={isActive}
    />)
    .toJSON();

  expect(tree).toMatchSnapshot();
});
