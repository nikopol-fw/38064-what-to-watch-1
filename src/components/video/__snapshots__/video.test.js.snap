// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Video correctly renders 1`] = `
<video
  autoPlay={true}
  height={175}
  loop={true}
  muted={true}
  poster="/img/https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm"
  preload="metadata"
  src="fantastic-beasts-the-crimes-of-grindelwald.jpg"
  width={280}
/>
`;
